Content.Server/Chemistry/ReagentEffects/PlantMetabolism/RobustHarvest.cs

```diff
@@ -31,21 +33,16 @@ public override void Effect(ReagentEffectArgs args)
             var plantHolder = args.EntityManager.System<PlantHolderSystem>();
             var random = IoCManager.Resolve<IRobustRandom>();
 
-            if (plantHolderComp.Seed.Potency < PotencyLimit)
+            if (plantHolderComp.PotencyBonus < PotencyLimit)
             {
                 plantHolder.EnsureUniqueSeed(args.SolutionEntity, plantHolderComp);
-                plantHolderComp.Seed.Potency = Math.Min(plantHolderComp.Seed.Potency + PotencyIncrease, PotencyLimit);
-
-                if (plantHolderComp.Seed.Potency > PotencySeedlessThreshold)
-                {
-                    plantHolderComp.Seed.Seedless = true;
-                }
+                plantHolderComp.PotencyBonus = Math.Min(plantHolderComp.PotencyBonus + PotencyIncrease, PotencyLimit);
             }
-            else if (plantHolderComp.Seed.Yield > 1 && random.Prob(0.1f))
+            else if ((float) plantHolderComp.Seed.Yield * plantHolderComp.YieldMod > 1f && random.Prob(YieldReductionProbability))
             {
                 // Too much of a good thing reduces yield
                 plantHolder.EnsureUniqueSeed(args.SolutionEntity, plantHolderComp);
-                plantHolderComp.Seed.Yield--;
+                plantHolderComp.YieldMod -= 1f / (float)plantHolderComp.Seed.Yield;
```

Comment:
This seems like it reduces all the yield in one step. Would it be better to gradually reduce yield and increase YieldReductionProbability?

Reply:
Ok, I'll admit, this is kinda hacky.
This is supposed to reduce the yieldmod by the equivalent of a single produce.
So if a plant has 5 yield, it should reduce it by 0.2, thus reducing the amount harvested by 1.

Reply:
This sounds fine when explained. Go ahead and add a comment explaining what the intended behavior is (which is to decrease yield by one plant unit).